### PR TITLE
[FIX] mail_plugin: do not allow to create a partner from notification email

### PR DESCRIPTION
**Pull request by std-odoo**
Purpose
=======
When we open an email that has been encapsulated in the notification
email, we still search the partner based on that email (which make no
sens). Instead, in that case, we show an error saying that it's the
notification email.

Task-3557545

### CODE REVIEW

File: addons/mail_plugin/controllers/mail_plugin.py [hunk 1]
```diff
@@ -140,7 +140,6 @@ def res_partner_get(self, email=None, name=None, partner_id=None, **kwargs):
         versions of the mail plugin but necessary for supporting older versions, only the route name is deprecated not
         the entire method.
         """
-
```
Comment (awa-odoo) [inline addons/mail_plugin/controllers/mail_plugin.py:143]: (Small note: Let's avoid adding unnecessary diff in old stable if possible)

File: addons/mail_plugin/controllers/mail_plugin.py [hunk 2]
```diff
@@ -152,10 +151,25 @@ def res_partner_get(self, email=None, name=None, partner_id=None, **kwargs):
...
+                },
+            }
+        else:
+            # Search for the partner based on the email.
+            # If multiple are found, take the first one.
+            partner = request.env['res.partner'].search(
+                ['|', ('email', 'in', [normalized_email, email]),
+                 ('email_normalized', '=', normalized_email)],
+                limit=1,
+            )
```
Comment (awa-odoo) [inline addons/mail_plugin/controllers/mail_plugin.py:172]: Would avoid putting this inside a "else", as it's a pre-condition, a bit like the "if not normalized_email" above.

In addition, it will allow to not modify that part, making it a smaller diff for an old stable.

File: addons/mail_plugin/controllers/mail_plugin.py [hunk 3]
```diff
@@ -140,7 +140,6 @@ def res_partner_get(self, email=None, name=None, partner_id=None, **kwargs):
         versions of the mail plugin but necessary for supporting older versions, only the route name is deprecated not
```
Comment (awa-odoo) [inline addons/mail_plugin/controllers/mail_plugin.py:140]: Sugg for the commit message:

> When an email is opened and is from the default notification mechanism,
i.e: coming from something like "notifications@mail.odoo.com", we don't want to
try to match the email address to an existing contact partner.
> 
> Instead, we now display a custom message on the mail addon side mentioning to
search the contact manually.

File: addons/mail_plugin/tests/test_controller.py [hunk 4]
```diff
@@ -187,3 +187,10 @@ def test_get_partner_no_email_returned_by_iap(self):
         )
         second_company_id = result["partner"]["company"]["id"]
         self.assertEqual(first_company_id, second_company_id, "Should not create a new company")
+
+    def test_get_partner_notification_email(self):
+        """The EMAIL FROM is the notification email."""
+        self.env["ir.config_parameter"].sudo().set_param("mail.default.from", "notification@example.com")
+        mock_iap_enrich = Mock()
+        result = self.mock_plugin_partner_get("Test partner", "notificaTION@EXAMPLE.COM", mock_iap_enrich)
+        self.assertEqual(result, {'partner': {'name': 'Notification', 'email': 'notification@example.com', 'enrichment_info': {'type': 'odoo_custom_error', 'info': 'This is your notification address. Search the Contact manually to link this email to a record.'}}})
```
Comment (awa-odoo) [inline addons/mail_plugin/tests/test_controller.py:196]: Can we multi-line this a bit so it's easier to read?

File: addons/mail_plugin/tests/test_controller.py [hunk 5]
```diff
@@ -187,3 +187,10 @@ def test_get_partner_no_email_returned_by_iap(self):
         )
         second_company_id = result["partner"]["company"]["id"]
         self.assertEqual(first_company_id, second_company_id, "Should not create a new company")
+
+    def test_get_partner_notification_email(self):
+        """The EMAIL FROM is the notification email."""
```
Comment (awa-odoo) [inline addons/mail_plugin/tests/test_controller.py:192]: Sugg: """ When the email_from is the server default from address, we return a custom message instead of trying to match a partner record """

File: addons/mail_plugin/tests/test_controller.py [hunk 6]
```diff
@@ -187,3 +187,10 @@ def test_get_partner_no_email_returned_by_iap(self):
         )
         second_company_id = result["partner"]["company"]["id"]
         self.assertEqual(first_company_id, second_company_id, "Should not create a new company")
+
+    def test_get_partner_notification_email(self):
```
Comment (awa-odoo) [inline addons/mail_plugin/tests/test_controller.py:191]: "test_get_partner_is_default_from" ?

**Review by awa-odoo — COMMENTED** [review]
Hello @std-odoo 

Same thing here, the usual naming suggestions and polishing remarks.

Thanks for your work & cheers!

File: addons/mail_plugin/controllers/mail_plugin.py (hunk 2 above)
Comment (std-odoo) [reply addons/mail_plugin/controllers/mail_plugin.py]: ah indeed, good catch

**Review by std-odoo — COMMENTED** [review]
@awa-odoo Hi :) 

I've done the changes

Cheers

**Comment by awa-odoo** [comment]
Hello @std-odoo 

Runbot seems mad for something unrelated, maybe needs a rebase if you have not already?

After that we are good to go, thanks for the quick response 🚀

**Comment by std-odoo** [comment]
@awa-odoo Re :) It's already on the latest 16.0 :/ I restarted the runbot

**Comment by std-odoo** [comment]
@awa-odoo Re :) Same here, the runbot is now green

**Comment by awa-odoo** [comment]
@robodoo r+
